plugins/source/azure/resources/services/eventhub/namespaces.go

```diff
@@ -1,242 +1,142 @@
+				Name:     "name",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Name"),
 			},
 			{
-				Name:        "network_rule_set",
-				Description: "Network rule set for a namespace.",
-				Type:        schema.TypeJSON,
-				Resolver:    resolveNamespaceNetworkRuleSet,
+				Name:     "type",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Type"),
 			},
 		},
+
 		Relations: []*schema.Table{
-			{
-				Name:          "azure_eventhub_namespace_encryption_key_vault_properties",
-				Description:   "KeyVaultProperties properties to configure keyVault Properties",
-				Resolver:      fetchEventhubNamespaceEncryptionKeyVaultProperties,
-				IgnoreInTests: true,
-				Columns: []schema.Column{
-					{
-						Name:        "namespace_cq_id",
-						Description: "Unique CloudQuery ID of azure_eventhub_namespaces table (FK)",
-						Type:        schema.TypeUUID,
-						Resolver:    schema.ParentIdResolver,
-					},
-					{
-						Name:        "key_name",
-						Description: "Name of the Key from KeyVault",
-						Type:        schema.TypeString,
-					},
-					{
-						Name:        "key_vault_uri",
-						Description: "Uri of KeyVault",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("KeyVaultURI"),
-					},
-					{
-						Name:        "key_version",
-						Description: "Key Version",
-						Type:        schema.TypeString,
-					},
-				},
-			},
+			networkRuleSets(),
 		},
 	}
 }
 
-// ====================================================================================================================
-//                                               Table Resolver Functions
-// ====================================================================================================================
+func fetchEventHubNamespaces(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	svc := meta.(*client.Client).Services().EventHub.Namespaces
 
-func fetchEventhubNamespaces(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
-	svc := meta.(*client.Client).Services().EventHub
 	response, err := svc.List(ctx)
+
 	if err != nil {
-		return diag.WrapError(err)
+		return errors.WithStack(err)
 	}
+
 	for response.NotDone() {
 		res <- response.Values()
 		if err := response.NextWithContext(ctx); err != nil {
-			return diag.WrapError(err)
+			return errors.WithStack(err)
 		}
 	}
-	return nil
-}
-func fetchEventhubNamespaceEncryptionKeyVaultProperties(_ context.Context, _ schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	namespace := parent.Item.(eventhub.EHNamespace)
-	if namespace.Encryption == nil || namespace.Encryption.KeyVaultProperties == nil {
-		return nil
-	}
-	res <- *namespace.Encryption.KeyVaultProperties
-	return nil
-}
 
-func resolveNamespaceNetworkRuleSet(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
```

Comment:
Moved this relation to a separate file `plugins/source/azure/resources/services/eventhub/network_rule_sets.go`